km63/Artemenko_Yaroslav/laboratory1.sql

```diff
@@ -1,2 +1,150 @@
 -- LABORATORY WORK 1
 -- BY Artemenko_Yaroslav
+/*---------------------------------------------------------------------------
+1.Створити схему даних з назвою – прізвище студента, у якій користувач може: 
+створювати таблиці та редагувати структуру таблиці
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
+CREATE USER artemenko IDENTIFIED BY artemenko 
+DEFAULT TABLESPACE "USERS"
+TEMPORARY TABLESPACE "TEMP";
+
+ALTER USER artemenko quota 100M on users;
+
+GRANT "connect" to artemenko;
+
+GRANT CREATE ANY TABLE to artemenko;
+GRANT ALTER ANY TABLE to artemenko;
+
+
+
+
+
+
+
+
+/*---------------------------------------------------------------------------
+2. Створити таблиці, у яких визначити поля та типи. Головні та зовнішні ключі 
+створювати окремо від таблиць використовуючи команди ALTER TABLE. 
+Людина має аккаунт Facebook.
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
+
+CREATE TABLE people(
+    people_name varchar2(30) NOT NULL
+    );
+ALTER TABLE people
+    ADD CONSTRAINT name_pk PRIMARY KEY (people_name);
+
+CREATE TABLE acc (
+    acc_id NUMBER(10) NOT NULL
+    );
+ALTER TABLE acc
+    ADD CONSTRAINT acc_pk PRIMARY KET (acc_id);
+    
+CREATE TABLE people_acc(
+    people_name_fk varchar(30) NOT NULL,
+    acc_id_fk NUMBER(10) NOT NULL,
+    friends varchar(30) NOT NULL
+    );
+ALTER TABLE people_acc
+    ADD CONSTRAINT people_acc_pk PRIMARY KEY (people_name_fk,acc_id_fk);
+ALTER TABLE people_acc
+    ADD CONSTRAINT people_fk FOREIGN KEY (people_name_fk) REFERENCES people(people_name);
+ALTER TABLE people_acc
+    ADD CONSTRAINT acc_fk FOREIGN KEY (acc_id_fk) REFERENCES acc(acc_if);
+    
```

Comment:
There should be 2 tables: Human and Facebook with one to many connection, because people can have soma accs in Facefook (f.e. your own acc, then u decided to create company and make the new acc and it's also yours or your main acc was hacked and u make the new one and in some time u recover hacked acc)
Table "Human" includes : 
- id_human - it will be primary key and auto_increment int
- Name_human - it will be begin from big letter and than little letters (not null)
- Surname_human - it will be the same like Name, also begins from big letter and other little letters (is null)
- Phone_human - it will be the field int where user will enter at first the code of his operator in country (Ukraine) it is 3 numbers and after code user write his own 7 numbers in hos phone number (is null)

Facebook includes:
- id_acc - it will be primary key and auto_increment int and it will be foreign key to id_human 
- email_acc - it will have compulsory "@" and different symbols which user has in his email (is null)
- date_create_acc - it will have form like date-month-year, where user can enter 1-31 in date, 1-12 in month and 2004-2018 in year (2004 because the Facebook was created in 2004) (not null)
- password_acc - it will have minimum 8 symbols and this symbols will include 1 or more big letters, 1 or more numbers and other symbols which user wanna has in his password (not null)